### PR TITLE
Prevent snap helpers from snapping until the user interacts with scro…

### LINKED ISSUE #4935: [CollectionView] setting snap to  Mandatory Snap Type only updates grid on Android
### Description
https://github.com/xamarin/Xamarin.Forms/pull/4414#issuecomment-452399726

On Android when you set it to Mandatory Snap Type the Grid will move to match the SnapPointsAlignment each time you change it. But that's the only scenario it does that. If you set it to Mandatory Single or do the same thing on iOS it doesn't update to match SnapPointsAlignment

collectionview

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

 By default, the LinearSnapHelper on Android snaps to the appropriate view immediately after being attached to the RecyclerView. However, the other platforms (and the single snap helpers) don't work this way. So to bring everything into line behaviorally, this change prevents the snap helpers on Android from snapping until the user interacts with the CollectionView.

### Issues Resolved ### 

- fixes #4935 

### API Changes ###

None

### Platforms Affected ### 

- Android

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by PureWeen** [comment]
Android doesn't respond to the snap type change as quickly as iOS if the changes are subtle to start with

**Comment by hartez** [comment]
@PureWeen Try it now. Should feel much closer to the iOS version.

File: Xamarin.Forms.Platform.Android/CollectionView/NongreedySnapHelper.cs [hunk 1]
```diff
@@ -8,16 +8,44 @@ internal abstract class NongreedySnapHelper : LinearSnapHelper
...
+				{
+					_recyclerView.ScrollChange -= RecyclerView_ScrollChange;
+				}
+			}
+
+			base.Dispose(disposing);
 		}
 
-		public override int[] CalculateScrollDistance(int velocityX, int velocityY)
+		void RecyclerView_ScrollChange(object sender, global::Android.Views.View.ScrollChangeEventArgs e)
```
[the remark this answers is not in the document]
Comment (rmarinho) [reply Xamarin.Forms.Platform.Android/CollectionView/NongreedySnapHelper.cs]: Yeah good catch .

File: Xamarin.Forms.Platform.Android/CollectionView/NongreedySnapHelper.cs [hunk 2]
```diff
@@ -8,16 +8,44 @@ internal abstract class NongreedySnapHelper : LinearSnapHelper
...
+		bool _disposed;
+		RecyclerView _recyclerView;
+
+		public override void AttachToRecyclerView(RecyclerView recyclerView)
 		{
-			CanSnap = true;
-			return base.FindTargetSnapPosition(layoutManager, velocityX, velocityY);
+			base.AttachToRecyclerView(recyclerView);
+
+			_recyclerView = recyclerView;
```
[the remark this answers is not in the document]
Comment (hartez) [reply Xamarin.Forms.Platform.Android/CollectionView/NongreedySnapHelper.cs]: @paymicro It's possible, but why do you think we need to?

**Pull request merged**